File: backend/app/schemas/assistant.py
from typing import List, Optional, Literal, Dict, Any
from pydantic import BaseModel, Field


class ChatMessage(BaseModel):
    role: Literal["user", "assistant", "system"]
    content: str


class ChatContext(BaseModel):
    current_path: str = Field(default="/", description="URL path of the page the user is currently on")
    selected_job_id: Optional[int] = Field(default=None, description="Current job ID if viewing a job detail page")
    role: Optional[str] = Field(default=None, description="User role if authenticated: candidate, employer, admin")


class EmbeddedCard(BaseModel):
    card_type: Literal["job", "tool", "action", "info"]
    title: str
    subtitle: Optional[str] = None
    url: str
    meta: Optional[Dict[str, Any]] = None


class AssistantChatRequest(BaseModel):
    messages: List[ChatMessage]
    context: Optional[ChatContext] = None


class AssistantChatResponse(BaseModel):
    reply: str
    suggested_cards: List[EmbeddedCard] = Field(default_factory=list)
    suggested_followups: List[str] = Field(default_factory=list)


class AssistantQuickSuggestion(BaseModel):
    label: str
    prompt: str
    category: Literal["job_search", "cv_help", "interview", "employer_jd", "tools", "general"]

File: frontend/src/lib/api/assistant.ts
import { apiClient } from "@/lib/axios";

export interface ChatMessage {
  role: "user" | "assistant" | "system";
  content: string;
}

export interface EmbeddedCard {
  card_type: "job" | "tool" | "action" | "info";
  title: string;
  subtitle?: string | null;
  url: string;
  meta?: Record<string, unknown> | null;
}

export interface AssistantChatResponse {
  reply: string;
  suggested_cards: EmbeddedCard[];
  suggested_followups: string[];
}

export interface AssistantQuickSuggestion {
  label: string;
  prompt: string;
  category: "job_search" | "cv_help" | "interview" | "employer_jd" | "tools" | "general";
}

export interface AssistantChatPayload {
  messages: ChatMessage[];
  context?: {
    current_path: string;
    selected_job_id?: number | null;
    role?: string | null;
  };
}

/** Send conversational chat to JobPortal AI Copilot */
export async function chatWithAssistant(payload: AssistantChatPayload): Promise<AssistantChatResponse> {
  const response = await apiClient.post<AssistantChatResponse>("/ai/assistant/chat", payload);
  return response.data;
}

/** Get context-aware quick prompt suggestions */
export async function getAssistantSuggestions(path: string = "/", role?: string): Promise<AssistantQuickSuggestion[]> {
  const response = await apiClient.get<AssistantQuickSuggestion[]>("/ai/assistant/suggestions", {
    params: { path, role },
  });
  return response.data;
}
